quickwit/quickwit-doc-mapper/src/routing_expression/mod.rs

```diff
@@ -323,8 +323,14 @@ impl Display for InnerRoutingExpr {
                     if index != 0 {
                         f.write_str(".")?;
                     }
-                    // TODO actually we should escape .
-                    f.write_str(part)?;
+                    if !part.contains('.') {
+                        f.write_str(part)?;
+                    } else {
+                        // TODO use std version once stabilized
+                        for subpart in itertools::Itertools::intersperse(part.split('.'), r"\.") {
+                            f.write_str(subpart)?;
+                        }
+                    }
```

Comment:
I'm  a bit lost. When is this block different from `part.replace(".", r"\.")`

Reply:
hum, never actually. It saves an allocation, but here we don't care